### ISSUE TITLE
typeahead onSelect function bug?

### ISSUE THREAD

**Issue by sh977218** [angular-ui/bootstrap#6258]
I'm not sure if this is bug or intentionally doing this.
when onSelect got called, the value in ng-model has already been replaced by the selected value. In this case, the onSelect cannot do anything to ng-model. What is the point of using onSelect?
I think the correct way should be changing the ng-model after user selected the value.

**Comment by ducksoupdev** [comment]
I agree, it is a shame that the ng-model is set before the onSelect callback is fired in order to compare the before and after values. Maybe an optional `beforeSelect` callback that is fired before the ng-model is set? Happy to submit a PR for it.

**Comment by sh977218** [comment]
I'm moving to use angualr, @ng-bootstrap/ng-bootstrap typeahead.  You can do one way binding of [ngModal], then (change) event emitter will act like beforeChange over there. If there is way to make ng-modal do one way binding, it will solve the problem.
